### PR TITLE
Updates MC documentation towards updates

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/15700

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [x] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
I could use a hint on:
```
ERROR 2021/02/19 11:28:15 [en] REF_NOT_FOUND: Ref "/ui-api/management-api/crd/releases.release.giantswarm.io": "/docs/src/content/general/management-clusters/index.md:28:86": page not found
```
as that looks correct to me, for https://docs.giantswarm.io/ui-api/management-api/crd/releases.release.giantswarm.io/

File: src/content/general/management-clusters/index.md [hunk 1]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
 
 Our workload clusters are versioned using [workload cluster releases]({{< relref "/general/releases" >}}). From a workload cluster point of view, upgrades are described in [cluster upgrades]({{< relref "/general/cluster-upgrades" >}}).
 
-When we publish a new Giant Swarm release, we create a new release custom resource and any new operator versions are deployed to the management cluster.
-We deploy new instances of the operators with the new version to avoid impacting existing workload clusters. These new operators will not become active until existing workload clusters are upgraded or a new cluster is created.
+When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io/" >}}) and any new operator versions are deployed to the management cluster.
```
Comment (rossf7) [inline src/content/general/management-clusters/index.md:28]:
```suggestion
When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.
```

I think it just needs the file extension.

File: src/content/general/management-clusters/index.md [hunk 2]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
...
+Changes to configuration are tested on our internal management clusters before being applied.
+
+This configuration system is being updated in Q1 2021, with a system based on the new [config-controller](https://github.com/giantswarm/config-controller/). This will remove the need for manual updates of configuration, remove the shared ConfigMap and Secret, and allow for versioning of configuration. Overall, aiming to further increase the safety, reliability, and scalability of configuration changes.
 
 ## Infrastructure upgrades
 
-The infrastructure including the management cluster nodes and supporting resources like bastion hosts are managed by our SRE team using Terraform.
+The management cluster infrastructure, such as the Kubernetes cluster itself, or other supporting resources such as bastion hosts, are managed using Terraform.
+
+Updates to our Terraform playbooks are rolled out by CI as needed, with changes to our Terraform playbooks being tested on internal management clusters before being deployed.
```
Comment (rossf7) [inline src/content/general/management-clusters/index.md:58]: I thought Terraform was modules and Ansible was playbooks? It's a good name but just not used to seeing it with TF.

File: src/content/general/management-clusters/index.md [hunk 3]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
...
 
-However, we plan to move more of this provisioning to operators, reusing the components that manage our workload clusters where possible. As they continuously reconcile the cluster to its desired state, operators are more dynamic than Terraform and other tools such as Kops.
+In 2021, we plan to start reusing our own workload clusters as the clusters that base management clusters, to both avoid having to maintain two methods of provisioning clusters, but also to gain the reliability benefits of our workload clusters.
```
Comment (rossf7) [inline src/content/general/management-clusters/index.md:60]: The bit about operators being more dynamic was taken from https://www.giantswarm.io/management-cluster

@teemow asked if we could consolidate the docs with the website page.
https://github.com/giantswarm/giantswarm/issues/15700#issuecomment-780029135

**Review by rossf7 — COMMENTED** [review]
@JosephSalisbury Looks good just a couple of rando thoughts.

@teemow Agree with Joe I think makes sense to merge this first and then your PR. Otherwise we'll get a ton of merge problems.

File: src/content/general/management-clusters/index.md (hunk 3 above)
Comment (rossf7) [reply src/content/general/management-clusters/index.md:60]: Fine to drop it from my POV if you think it isn't needed.

File: src/content/general/management-clusters/index.md (hunk 1 above)
Comment (marians) [inline src/content/general/management-clusters/index.md:28]:
```suggestion
When we publish a new workload cluster release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io/" >}}) and any new operator versions are deployed to the management cluster.
```

File: src/content/general/management-clusters/index.md (hunk 2 above)
Comment (JosephSalisbury) [reply src/content/general/management-clusters/index.md:58]: yeah you're fucking right, i'm confusing the two

File: src/content/general/management-clusters/index.md (hunk 3 above)
Comment (JosephSalisbury) [reply src/content/general/management-clusters/index.md:60]: i'll take this into an issue in biscuit, because we can be a bit more aligned here, sure

Comment (JosephSalisbury) [reply src/content/general/management-clusters/index.md:60]: (see https://github.com/orgs/giantswarm/projects/125#card-55271871)

File: src/content/general/management-clusters/index.md (hunk 1 above)
Comment (JosephSalisbury) [reply src/content/general/management-clusters/index.md]: lol, apparently so. thanks!

Comment (JosephSalisbury) [reply src/content/general/management-clusters/index.md]: i don't know how, but this got resolved somewhere :D

File: src/content/general/management-clusters/index.md [hunk 4]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
 
 Our workload clusters are versioned using [workload cluster releases]({{< relref "/general/releases" >}}). From a workload cluster point of view, upgrades are described in [cluster upgrades]({{< relref "/general/cluster-upgrades" >}}).
 
-When we publish a new Giant Swarm release, we create a new release custom resource and any new operator versions are deployed to the management cluster.
-We deploy new instances of the operators with the new version to avoid impacting existing workload clusters. These new operators will not become active until existing workload clusters are upgraded or a new cluster is created.
+When we publish a new Giant Swarm release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.
```
Comment (marians) [inline src/content/general/management-clusters/index.md:28]:
```suggestion
When we publish a new workload cluster release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.
```

Comment (JosephSalisbury) [reply src/content/general/management-clusters/index.md]: i don't know how git works, commited 🚀

File: src/content/general/management-clusters/index.md [hunk 5]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
 
 Our workload clusters are versioned using [workload cluster releases]({{< relref "/general/releases" >}}). From a workload cluster point of view, upgrades are described in [cluster upgrades]({{< relref "/general/cluster-upgrades" >}}).
 
-When we publish a new Giant Swarm release, we create a new release custom resource and any new operator versions are deployed to the management cluster.
-We deploy new instances of the operators with the new version to avoid impacting existing workload clusters. These new operators will not become active until existing workload clusters are upgraded or a new cluster is created.
+When we publish a new workload cluster release, we create a new [Release Custom Resource]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) and any new operator versions are deployed to the management cluster.
```
Comment (marians) [inline src/content/general/management-clusters/index.md:28]:
```suggestion
When we publish a new workload cluster release, we create a new [Release]({{< relref "/ui-api/management-api/crd/releases.release.giantswarm.io.md" >}}) custom resource and any new operator versions are deployed to the management cluster.
```

No need to capitalize custom resource

File: src/content/general/management-clusters/index.md [hunk 6]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
...
+
+We deploy new instances of the operators with this new version, via the [release-operator](https://github.com/giantswarm/release-operator/) to avoid impacting existing workload clusters. These new operators will not become active until existing workload clusters are upgraded or a new cluster is created.
 
 ## Management cluster-only components
 
-The components that are not part of a workload cluster release -- like our monitoring stack -- are deployed via app collections. We run the latest version of each component and the collection is managed by our legacy deployment component called draughtsman (which we plan to retire in Q2 2021).
+There are a number of components that are not a part of a workload cluster release, such as our monitoring stack, or web UI. These components are deployed exclusively to the management cluster, and do not affect workload clusters.
+
+These components are deployed via app collections, which are a set of App Custom Resources, which are ultimately deployed via the App Platform. This collection contains the latest tagged version of each component, and is managed by our deployment component [draughtsman](https://github.com/giantswarm/draughtsman/).
```
Comment (marians) [inline src/content/general/management-clusters/index.md:36]:
```suggestion
These components are deployed via app collections, which are a set of App custom resources, which are ultimately deployed via the App Platform. This collection contains the latest tagged version of each component, and is managed by our deployment component [draughtsman](https://github.com/giantswarm/draughtsman/).
```

File: src/content/general/management-clusters/index.md [hunk 7]
```diff
@@ -25,21 +25,36 @@ As well as operators, our management clusters run supporting components such as
...
+These management cluster-only components are tested on our internal management clusters before releases are cut.
+
+We plan on replacing draughtsman in Q2 2021, with a new component that more natively synchronises app collections to a management cluster, mostly to improve the scalability of the system.
 
 ## Configuration changes
 
-From time to time, we also need to update the configuration for management clusters. These changes are also tested on our internal management clusters before being applied.
+Periodically, we need to update the configuration for management clusters.
+
+This configuration is currently held in a ConfigMap and a Secret stored on the Management Cluster. When updates are required, these resources are updated manually using our current tooling, from our central configuration repository.
```
Comment (marians) [inline src/content/general/management-clusters/index.md:46]:
```suggestion
This configuration is currently held in a ConfigMap and a Secret stored on the management cluster. When updates are required, these resources are updated manually using our current tooling, from our central configuration repository.
```

**Review by marians — APPROVED** [review]

**Pull request merged**